### PR TITLE
Workaround for Non CDBs database

### PR DESCRIPTION
**Pull request by hallelujah**
Closes https://issues.jboss.org/browse/THREESCALE-2120

When a customer does not use Containerized Database for Oracle

This is a workaround as non-containierized database are not supported.

### CODE REVIEW

File: amp/system-oracle/workaround-non-cdb-2.4-upgrade/database.rb [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+          result = nil
+          pool.with_connection do |connection|
+            result = connection.select_value(sql_for_readiness).tap { pool.disconnect! }
+          end
+          result.to_s == '1'
+        end
+
+        def sql_for_readiness
+          if oracle?
+            "SELECT 1 FROM DUAL"
```
Comment (hallelujah) [inline amp/system-oracle/workaround-non-cdb-2.4-upgrade/database.rb]: Probably better to modify a bit https://github.com/3scale/porta/blob/2.4.0/lib/system/database.rb#L58

by

```ruby
 "SELECT 1 FROM v$pdbs WHERE name COLLATE BINARY_CI = '#{connection_config[:database]}' AND open_mode = 'READ WRITE'"
```

select open_mode from v$database

Comment (hallelujah) [reply amp/system-oracle/workaround-non-cdb-2.4-upgrade/database.rb]: Not tested and not enough time for that. :/

**Comment by bravomail** [comment]
I just wanted to add that recommended SQL test string for Oracle (used industry wide) is : "SELECT 1 FROM DUAL".

**Comment by hallelujah** [comment]
@bravomail that does not test if the database can receive queries right?

File: amp/system-oracle/workaround-non-cdb-2.4-upgrade/database.rb (hunk 1 above)
Comment (guicassolato) [reply amp/system-oracle/workaround-non-cdb-2.4-upgrade/database.rb]: https://github.com/3scale/porta/pull/766

**Comment by hallelujah** [comment]
Merging, tested by  @3scale/qe

**Pull request merged**